plugins/modules/zos_zfs_resize.py

```diff
@@ -556,24 +601,33 @@ def run_module():
     tmp_file = ""
     trace_uss = True
     trace_destination_created = True
-    trace_type = ""
+    is_valid = True
+    msg_trace = ""
 
     if trace_destination is not None:
         if data_set.is_data_set(data_set.extract_dsname(trace_destination)):
             if data_set.is_member(trace_destination):
                 if not data_set.DataSet.data_set_exists(data_set.extract_dsname(trace_destination)):
                     trace_destination_created = create_trace_dataset(name=trace_destination, member=True)
+                else:
+                    is_valid, msg_trace = validate_information_dataset(dataset=trace_destination)
             else:
                 if not (data_set.DataSet.data_set_exists(trace_destination)):
                     trace_destination_created = create_trace_dataset(name=trace_destination, member=False)
                 else:
-                    trace_type = data_set.DataSet.data_set_type(trace_destination)
+                    is_valid, msg_trace = validate_information_dataset(dataset=trace_destination)
             trace_uss = False
         else:
             trace_destination = better_arg_parser.BetterArgHandler.fix_local_path(trace_destination)
             trace_uss = True
         tmp_file = trace_destination
 
+    if not is_valid:
```

Comment:
Would recommend to rename this variable from is_valid to is_valid_destination or is_valid_trace_destination